### PR TITLE
Ignore static methods when finding property accessors

### LINKED ISSUE #1458: Ignore static methods when introspecting properties
#### Summary

See the discussion starting here: https://github.com/joel-costigliola/assertj-core/pull/1360#issuecomment-470200904

### PR DESCRIPTION
**Pull request by wilkinsona**
Previously, when looking for property accessors, Introspection would consider a static method with a name in the required form as being an accessor for a property. Where such a static method existed, this was likely to result in the wrong value being considered as the static method is unlikely to return the correct value for an instance's property.

This PR updates Introspection so that static methods are ignored when looking for property accessors.

#### Check List:
* Fixes #1458 
* Unit tests : YES
* Javadoc with a code example (API only) : NA

### CODE REVIEW

**Comment by joel-costigliola** [comment]
Thanks @wilkinsona, I will review the PR shortly and likely cut another service pack release after it

File: src/main/java/org/assertj/core/util/introspection/Introspection.java [hunk 1]
```diff
@@ -77,14 +77,15 @@ private static Method findGetter(String propertyName, Object target) {
     String capitalized = propertyName.substring(0, 1).toUpperCase(ENGLISH) + propertyName.substring(1);
     // try to find getProperty
     Method getter = findMethod("get" + capitalized, target);
-    if (getter != null) return getter;
+    if (getter != null && !Modifier.isStatic(getter.getModifiers())) return getter;
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/util/introspection/Introspection.java]: let' s extract a static `isValidGetter(Method method)` method to capture the `getter != null && !Modifier.isStatic(getter.getModifiers())` repeated 3 times, will be more readable too.

File: src/test/java/org/assertj/core/util/introspection/PropertyOrFieldSupport_getValueOf_Test.java [hunk 2]
```diff
@@ -151,6 +151,27 @@ public void should_extract_single_value_from_maps_by_key() {
...
+    Object value = propertyOrFieldSupport.getValueOf("city", new StaticPropertyEmployee());
+
+    assertThat(value).isEqualTo("New York");
+  }
+
+  @Test
+  public void should_extract_field_value_if_only_static_is_method_matches_name() {
+    Object value = propertyOrFieldSupport.getValueOf("tall", new StaticBooleanPropertyEmployee());
+
+    assertThat(value).isEqualTo(false);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/util/introspection/PropertyOrFieldSupport_getValueOf_Test.java]: use `isFalse()`

**Comment by joel-costigliola** [comment]
@wilkinsona I have finished commenting the PR.

File: src/main/java/org/assertj/core/util/introspection/Introspection.java (hunk 1 above)
Comment (wilkinsona) [reply src/main/java/org/assertj/core/util/introspection/Introspection.java]: Thanks for the suggestions. I've pushed a commit to do this.

File: src/test/java/org/assertj/core/util/introspection/PropertyOrFieldSupport_getValueOf_Test.java (hunk 2 above)
Comment (wilkinsona) [reply src/test/java/org/assertj/core/util/introspection/PropertyOrFieldSupport_getValueOf_Test.java]: I don't think I can as `value` is an `Object` and there's no `isFalse()` method available on `ObjectAssert`.

Comment (joel-costigliola) [reply src/test/java/org/assertj/core/util/introspection/PropertyOrFieldSupport_getValueOf_Test.java]: yes, that's true

**Comment by joel-costigliola** [comment]
Integrated thanks @wilkinsona!

**Pull request closed without merging**